control/controlclient/direct.go

```diff
@@ -421,8 +421,12 @@ func (c *Direct) doLogin(ctx context.Context, opt loginOpt) (mustRegen bool, new
 	if res.StatusCode != 200 {
 		msg, _ := ioutil.ReadAll(res.Body)
 		res.Body.Close()
-		return regen, opt.URL, fmt.Errorf("register request: http %d: %.200s",
+		err := fmt.Errorf("register request: http %d: %.200s",
```

Comment:
This is an ugly message to be popped up in a Windows dialog box.

We control the server, so we should instead pass back the exact error message we want to show.

Reply:
Done. I kept some sensible max length limit.

After this change I'll scrub through the control server, making sure that error paths that return 4XX return something sensible.